org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java

```diff
@@ -63,7 +63,9 @@ protected void recursiveFindRpcMethods(Object current, Set<Class<?>> visited, Se
 					Method method = methodInfo.method;
 					Object[] arguments = this.getArguments(method, arg);
 					return (CompletableFuture<Object>) method.invoke(current, arguments);
-				} catch (InvocationTargetException | IllegalAccessException e) {
+				} catch (InvocationTargetException e) {
+					throw new RuntimeException(e.getCause());
```

Comment:
This will now loose information, e.g. assuming the InvocationTargetException's cause is an IllegalAccessException, the thrown RuntimeException would no longer allow to distinguish the two scenarios.

The use-case you describe in the ticket can be resolved by traversing the causes of the caught RuntimeException in your custom exception handler. You might want to look into  a solution based on `com.google.common.base.Throwables.getCausalChain(Throwable)` or `com.google.common.base.Throwables.getRootCause(Throwable)`